code/modules/admin/verbs/selectequipment.dm

```diff
@@ -0,0 +1,164 @@
+/client/proc/cmd_select_equipment(mob/M in GLOB.mob_list)
+	set category = "Admin.Events"
+	set name = "Select equipment but better"
+
+
+	var/datum/select_equipment/ui  = new(usr, M)
+	ui.ui_interact(usr)
+
+/datum/select_equipment
+	var/client/user
+	var/mob/target
+
+	var/dummy_key
+	var/mob/living/carbon/human/dummy/dummy
+
+	var/static/list/cached_outfits
+	var/static/list/cached_custom_outfits = list()
+	var/datum/outfit/selected_outfit = /datum/outfit
+
+/datum/select_equipment/New(usr, mob/M)
+	user = CLIENT_FROM_VAR(usr)
+
+	if(!(ishuman(M) || isobserver(M)))
+		alert("Invalid mob")
+		return
+	target = M
+
+/datum/select_equipment/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SelectEquipment", "Select Equipment")
+		ui.open()
+
+/datum/select_equipment/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/select_equipment/ui_status(mob/user, datum/ui_state/state)
+	if(QDELETED(target))
+		return UI_CLOSE
+	return ..()
+
+/datum/select_equipment/ui_close(mob/user)
+	clear_human_dummy(dummy_key)
+	qdel(src)
+
+/datum/select_equipment/proc/init_dummy()
+	dummy_key = "selectequipmentUI_[target]"
+	dummy = generate_or_wait_for_human_dummy(dummy_key)
+	var/mob/living/carbon/C = target
+	if(istype(C))
+		C.dna.transfer_identity(dummy)
+		dummy.updateappearance()
+
+	unset_busy_human_dummy(dummy_key)
+	return
+
+/datum/select_equipment/proc/make_outfit_entries(list/L)
+	var/list/entries = list()
+	for(var/path in L)
+		var/datum/outfit/O = path
+		entries[path] = initial(O.name)
+	return sortAssocList(entries)
+
+//GLOB.custom_outfits lists outfit *objects* so we'll need to do some custom handling for it
+/datum/select_equipment/proc/make_custom_outfit_entries(list/L)
+	var/list/entries = list()
+	for(var/datum/outfit/O in L)
+		cached_custom_outfits[O.name] = O
+		entries[O.name] = O.name //it's either this or special handling on the UI side
+	return sortAssocList(entries)
+
+/datum/select_equipment/ui_static_data(mob/user)
+	var/list/data = list()
+	if(!dummy)
+		init_dummy()
+
+	var/datum/preferences/prefs
+	if(target.client)
+		prefs = target.client.prefs
+	var/icon/dummysprite = get_flat_human_icon(null, prefs=prefs, dummy_key = dummy_key, outfit_override = selected_outfit)
+	data["icon64"] = icon2base64(dummysprite)
+
+	if(!cached_outfits)
+		//the assoc keys here will turn into Tabs in the UI, so make sure to name them well
+		cached_outfits = list()
+		cached_outfits["General"] = list(/datum/outfit = "Naked") + make_outfit_entries(subtypesof(/datum/outfit) - typesof(/datum/outfit/job) - typesof(/datum/outfit/plasmaman))
+		cached_outfits["Jobs"] = make_outfit_entries(typesof(/datum/outfit/job))
+		cached_outfits["Plasmamen Outfits"] = make_outfit_entries(typesof(/datum/outfit/plasmaman))
```

Comment:
These need to be nested inside a list so they are sent to the data as a mappable list and not an object you need to get the keys for and do annoying access with

Reply:
Not sure if you're talking about the categories or the outfit entries so I'll address both:

I think it's better for `cached_outfits` to be sent as an object as I can actually use the object methods as both keys and names for the Tabs as seen [here](https://github.com/tgstation/tgstation/pull/57673/files#diff-0931ef83d6077b6246c7d86ee4650436bea50aa7397bc02f5ba86145d5d9f49aR56-R57) - Unless doing so is cursed instead of clever (like I thought it was when I did it)

As for the outfit entries, the main reason I made them assoc lists in the first place is because sorting them in a list of lists would be both hard and ugly. Though I've still gotta find a way to implement favorites so I'll probably end up needing to implement custom sorting either way. ~Brain hurty please help~

Reply:
oh my god, what are you doing